File: acode-mac/ACode/Sources/Views/Settings/ProviderSettingsView.swift
import SwiftUI

/// Provider 管理设置页 — 原生 Form/List 风格，支持添加/编辑/删除/切换
struct ProviderSettingsView: View {
    @EnvironmentObject var appState: AppState
    let tool: String
    let toolName: String

    @State private var showAddSheet = false
    @State private var showPresetSheet = false
    @State private var editingProvider: Provider?

    private var toolProviders: [Provider] {
        appState.providers.filter { $0.tool == tool }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 20) {
            // 供应商列表
            SettingsSection(title: "已配置 (\(toolProviders.count))") {
                if toolProviders.isEmpty {
                    HStack {
                        Spacer()
                        VStack(spacing: 6) {
                            Image(systemName: "tray")
                                .font(.system(size: 24))
                                .foregroundColor(.secondary.opacity(0.4))
                            Text("暂无供应商")
                                .font(.system(size: 13))
                                .foregroundColor(.secondary)
                            Text("点击下方按钮添加")
                                .font(.system(size: 11))
                                .foregroundColor(.secondary.opacity(0.7))
                        }
                        .padding(.vertical, 24)
                        Spacer()
                    }
                } else {
                    ForEach(toolProviders) { provider in
                        ProviderRow(
                            provider: provider,
                            onEdit: { editingProvider = provider },
                            onDelete: { appState.deleteProvider(id: provider.id) },
                            onSwitch: { appState.switchProvider(id: provider.id) }
                        )

                        if provider.id != toolProviders.last?.id {
                            Divider()
                                .padding(.horizontal, 14)
                        }
                    }
                }
            }

            // 操作按钮
            HStack(spacing: 10) {
                Button(action: { showAddSheet = true }) {
                    Label("手动添加", systemImage: "plus")
                }
                .buttonStyle(.borderedProminent)
                .controlSize(.regular)

                Button(action: { showPresetSheet = true }) {
                    Label("从预设添加", systemImage: "list.bullet")
                }
                .buttonStyle(.bordered)
                .controlSize(.regular)

                Spacer()

                Text("点击圆点切换激活")
                    .font(.system(size: 11))
                    .foregroundColor(.secondary.opacity(0.5))
            }

            Spacer()
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .sheet(isPresented: $showAddSheet) {
            ProviderFormSheet(tool: tool, toolName: toolName, provider: nil) {
                appState.loadProviders()
            }
            .environmentObject(appState)
        }
        .sheet(item: $editingProvider) { provider in
            ProviderFormSheet(tool: tool, toolName: toolName, provider: provider) {
                appState.loadProviders()
            }
            .environmentObject(appState)
        }
        .sheet(isPresented: $showPresetSheet) {
            PresetPickerSheet(tool: tool) {
                appState.loadProviders()
            }
            .environmentObject(appState)
        }
    }
}

// MARK: - Provider Row (原生列表行)

struct ProviderRow: View {
    let provider: Provider
    let onEdit: () -> Void
    let onDelete: () -> Void
    let onSwitch: () -> Void

    @State private var showDeleteConfirm = false

    var body: some View {
        HStack(spacing: 12) {
            // 激活状态指示（点击切换）
            Button(action: onSwitch) {
                Image(systemName: provider.isActive ? "checkmark.circle.fill" : "circle")
                    .font(.system(size: 18))
                    .foregroundColor(provider.isActive ? .green : .secondary.opacity(0.3))
            }
            .buttonStyle(.plain)
            .help(provider.isActive ? "当前激活" : "切换到此供应商")

            // 信息
            VStack(alignment: .leading, spacing: 3) {
                HStack(spacing: 6) {
                    Text(provider.name)
                        .font(.system(size: 14, weight: provider.isActive ? .medium : .regular))
                    if provider.isActive {
                        Text("使用中")
                            .font(.system(size: 10, weight: .medium))
                            .foregroundColor(.green)
                            .padding(.horizontal, 5)
                            .padding(.vertical, 2)
                            .background(Color.green.opacity(0.1), in: RoundedRectangle(cornerRadius: 4))
                    }
                }

                HStack(spacing: 6) {
                    if !provider.model.isEmpty {
                        Text(provider.model)
                            .font(.system(size: 12))
                            .foregroundColor(.secondary)
                    }
                    Text(provider.maskedApiKey)
                        .font(.system(size: 11, design: .monospaced))
                        .foregroundColor(.secondary.opacity(0.5))
                }
            }

            Spacer()

            // 编辑/删除
            Button("编辑", action: onEdit)
                .buttonStyle(.bordered)
                .controlSize(.small)

            Button(action: { showDeleteConfirm = true }) {
                Image(systemName: "trash")
                    .font(.system(size: 12))
                    .foregroundColor(.red.opacity(0.6))
            }
            .buttonStyle(.plain)
            .confirmationDialog("确认删除", isPresented: $showDeleteConfirm) {
                Button("删除", role: .destructive, action: onDelete)
                Button("取消", role: .cancel) {}
            } message: {
                Text("确定要删除供应商 \"\(provider.name)\" 吗？")
            }
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 10)
    }
}

// MARK: - Provider Add/Edit Form Sheet

struct ProviderFormSheet: View {
    @EnvironmentObject var appState: AppState
    @Environment(\.dismiss) var dismiss

    let tool: String
    let toolName: String
    let provider: Provider? // nil = 新建模式
    let onSave: () -> Void

    @State private var name = ""
    @State private var apiKey = ""
    @State private var apiBase = ""
    @State private var model = ""
    @State private var extraEnv = "{}"
    @State private var notes = ""

    // Claude 专用
    @State private var haikuModel = ""
    @State private var sonnetModel = ""
    @State private var opusModel = ""

    @State private var errorMessage: String?

    private var isEditing: Bool { provider != nil }

    var body: some View {
        VStack(spacing: 0) {
            // 标题
            HStack {
                Text(isEditing ? "编辑 \(toolName) 供应商" : "添加 \(toolName) 供应商")
                    .font(.headline)
                Spacer()
                Button(action: { dismiss() }) {
                    Image(systemName: "xmark.circle.fill")
                        .foregroundColor(.secondary)
                }
                .buttonStyle(.plain)
            }
            .padding()

            Divider()

            // 表单
            Form {
                Section("基本信息") {
                    TextField("名称 *", text: $name, prompt: Text("如：官方 Claude"))
                    SecureField("API Key *", text: $apiKey, prompt: Text("sk-xxx"))
                    TextField("API 端点", text: $apiBase, prompt: Text("空=使用官方默认端点"))
                }

                Section("模型设置") {
                    TextField("主模型", text: $model, prompt: Text("空=使用默认模型"))

                    // Claude 专用多模型字段
                    if tool == "claude_code" {
                        TextField("Haiku 模型", text: $haikuModel, prompt: Text("空=默认"))
                        TextField("Sonnet 模型", text: $sonnetModel, prompt: Text("空=默认"))
                        TextField("Opus 模型", text: $opusModel, prompt: Text("空=默认"))
                    }
                }

                Section("高级") {
                    TextField("额外环境变量 (JSON)", text: $extraEnv, prompt: Text("{}"), axis: .vertical)
                        .font(.system(.body, design: .monospaced))
                        .lineLimit(3...6)

                    TextField("备注", text: $notes, prompt: Text("可选"))
                }

                // 错误提示
                if let error = errorMessage {
                    Section {
                        Text(error)
                            .foregroundColor(.red)
                            .font(.caption)
                    }
                }
            }
            .formStyle(.grouped)

            Divider()

            // 操作按钮
            HStack {
                Spacer()
                Button("取消") { dismiss() }
                    .keyboardShortcut(.cancelAction)
                Button(isEditing ? "保存" : "添加") { save() }
                    .buttonStyle(.borderedProminent)
                    .keyboardShortcut(.defaultAction)
                    .disabled(name.trimmingCharacters(in: .whitespaces).isEmpty ||
                              apiKey.trimmingCharacters(in: .whitespaces).isEmpty)
            }
            .padding()
        }
        .frame(width: 500, height: 600)
        .onAppear {
            if let p = provider {
                name = p.name
                apiKey = p.apiKey
                apiBase = p.apiBase
                model = p.model
                extraEnv = p.extraEnv
                notes = p.notes ?? ""

                // 解析 Claude 多模型
                if let extra = p.extraEnvDict {
                    haikuModel = extra["ANTHROPIC_DEFAULT_HAIKU_MODEL"] ?? ""
                    sonnetModel = extra["ANTHROPIC_DEFAULT_SONNET_MODEL"] ?? ""
                    opusModel = extra["ANTHROPIC_DEFAULT_OPUS_MODEL"] ?? ""
                }
            }
        }
    }

    private func save() {
        errorMessage = nil

        let formData = ProviderFormData(
            name: name.trimmingCharacters(in: .whitespaces),
            tool: tool,
            apiKey: apiKey.trimmingCharacters(in: .whitespaces),
            apiBase: apiBase.trimmingCharacters(in: .whitespaces),
            model: model,
            extraEnv: extraEnv,
            icon: provider?.icon,
            iconColor: provider?.iconColor,
            notes: notes.isEmpty ? nil : notes,
            category: provider?.category,
            presetId: provider?.presetId,
            haikuModel: haikuModel,
            sonnetModel: sonnetModel,
            opusModel: opusModel
        )

        do {
            if let existingProvider = provider {
                _ = try appState.providerService.updateProvider(id: existingProvider.id, data: formData)
            } else {
                _ = try appState.providerService.createProvider(formData)
            }
            onSave()
            dismiss()
        } catch {
            errorMessage = error.localizedDescription
        }
    }
}

// MARK: - Preset Picker Sheet

struct PresetPickerSheet: View {
    @EnvironmentObject var appState: AppState
    @Environment(\.dismiss) var dismiss

    let tool: String
    let onSave: () -> Void

    @State private var selectedPreset: ProviderPreset?
    @State private var apiKey = ""
    @State private var errorMessage: String?

    private var filteredPresets: [ProviderPreset] {
        ProviderPreset.presets.filter { $0.tool == tool }
    }

    var body: some View {
        VStack(spacing: 0) {
            HStack {
                Text("从预设添加")
                    .font(.headline)
                Spacer()
                Button(action: { dismiss() }) {
                    Image(systemName: "xmark.circle.fill")
                        .foregroundColor(.secondary)
                }
                .buttonStyle(.plain)
            }
            .padding()

            Divider()

            List {
                ForEach(filteredPresets) { preset in
                    HStack {
                        VStack(alignment: .leading) {
                            Text(preset.name)
                                .font(.system(size: 13, weight: .medium))
                            Text(preset.apiBase.isEmpty ? "官方默认端点" : preset.apiBase)
                                .font(.system(size: 11))
                                .foregroundColor(.secondary)
                        }
                        Spacer()
                        Text(preset.defaultModel)
                            .font(.system(size: 11))
                            .foregroundColor(.secondary)

                        if selectedPreset?.id == preset.id {
                            Image(systemName: "checkmark")
                                .foregroundColor(.green)
                        }
                    }
                    .contentShape(Rectangle())
                    .onTapGesture { selectedPreset = preset }
                    .listRowBackground(
                        selectedPreset?.id == preset.id
                            ? Color(nsColor: .separatorColor).opacity(0.3)
                            : Color.clear
                    )
                }
            }
            .listStyle(.inset)

            if selectedPreset != nil {
                Divider()
                HStack {
                    SecureField("API Key *", text: $apiKey, prompt: Text("输入 API Key"))
                    Button("添加") { addFromPreset() }
                        .buttonStyle(.borderedProminent)
                        .disabled(apiKey.trimmingCharacters(in: .whitespaces).isEmpty)
                }
                .padding()
            }

            if let error = errorMessage {
                Text(error)
                    .foregroundColor(.red)
                    .font(.caption)
                    .padding(.horizontal)
            }
        }
        .frame(width: 450, height: 400)
    }

    private func addFromPreset() {
        guard let preset = selectedPreset else { return }
        errorMessage = nil

        let formData = ProviderFormData(
            name: preset.name,
            tool: preset.tool,
            apiKey: apiKey.trimmingCharacters(in: .whitespaces),
            apiBase: preset.apiBase,
            model: preset.defaultModel,
            icon: preset.icon,
            iconColor: preset.iconColor,
            presetId: preset.id
        )

        do {
            _ = try appState.providerService.createProvider(formData)
            onSave()
            dismiss()
        } catch {
            errorMessage = error.localizedDescription
        }
    }
}

File: acode-windows/lib/views/settings/provider_settings_view.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart' as p;

import '../../app/app_state.dart';
import '../../models/provider.dart';
/// Provider 管理设置页
class ProviderSettingsView extends StatefulWidget {
  final String tool;
  final String toolName;

  const ProviderSettingsView({
    super.key,
    required this.tool,
    required this.toolName,
  });

  @override
  State<ProviderSettingsView> createState() => _ProviderSettingsViewState();
}

class _ProviderSettingsViewState extends State<ProviderSettingsView> {
  List<Provider> get _toolProviders {
    final appState = p.Provider.of<AppState>(context, listen: true);
    return appState.providers.where((pr) => pr.tool == widget.tool).toList();
  }

  @override
  Widget build(BuildContext context) {
    final appState = p.Provider.of<AppState>(context, listen: true);
    final isDark = Theme.of(context).brightness == Brightness.dark;
    final cardColor = isDark ? const Color(0xFF2D2D2D) : const Color(0xFFF5F5F5);

    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        // 已配置列表
        Row(
          children: [
            Text(
              '已配置 (${_toolProviders.length})',
              style: TextStyle(fontSize: 13, fontWeight: FontWeight.w600, color: Colors.grey[600]),
            ),
            const Spacer(),
            Text(
              '点击左侧圆点切换激活',
              style: TextStyle(fontSize: 11, color: Colors.grey[500]),
            ),
          ],
        ),
        const SizedBox(height: 8),

        if (_toolProviders.isEmpty)
          Container(
            padding: const EdgeInsets.symmetric(vertical: 20),
            decoration: BoxDecoration(
              color: cardColor,
              borderRadius: BorderRadius.circular(8),
            ),
            child: Center(
              child: Column(
                children: [
                  Text('暂无供应商', style: TextStyle(color: Colors.grey[500])),
                  const SizedBox(height: 4),
                  Text('点击下方按钮添加', style: TextStyle(fontSize: 12, color: Colors.grey[400])),
                ],
              ),
            ),
          )
        else
          ...List.generate(_toolProviders.length, (i) {
            final provider = _toolProviders[i];
            return Padding(
              padding: const EdgeInsets.only(bottom: 8),
              child: _ProviderRow(
                provider: provider,
                onSwitch: () => appState.switchProvider(provider.id!),
                onEdit: () => _showEditDialog(provider),
                onDelete: () => _showDeleteConfirm(provider),
              ),
            );
          }),

        const SizedBox(height: 16),

        // 操作按钮
        Row(
          children: [
            ElevatedButton.icon(
              onPressed: () => _showAddDialog(),
              icon: const Icon(Icons.add, size: 16),
              label: const Text('手动添加供应商'),
            ),
            const SizedBox(width: 8),
            OutlinedButton.icon(
              onPressed: () => _showPresetDialog(),
              icon: const Icon(Icons.list, size: 16),
              label: const Text('从预设快速添加'),
            ),
          ],
        ),
      ],
    );
  }

  void _showAddDialog() {
    showDialog(
      context: context,
      builder: (ctx) => _ProviderFormDialog(
        tool: widget.tool,
        toolName: widget.toolName,
        onSave: () {
          p.Provider.of<AppState>(context, listen: false).loadProviders();
        },
      ),
    );
  }

  void _showEditDialog(Provider provider) {
    showDialog(
      context: context,
      builder: (ctx) => _ProviderFormDialog(
        tool: widget.tool,
        toolName: widget.toolName,
        provider: provider,
        onSave: () {
          p.Provider.of<AppState>(context, listen: false).loadProviders();
        },
      ),
    );
  }

  void _showDeleteConfirm(Provider provider) {
    showDialog(
      context: context,
      builder: (ctx) => AlertDialog(
        title: const Text('确认删除'),
        content: Text('确定要删除供应商 "${provider.name}" 吗？'),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(ctx),
            child: const Text('取消'),
          ),
          TextButton(
            onPressed: () {
              Navigator.pop(ctx);
              p.Provider.of<AppState>(context, listen: false).deleteProvider(provider.id!);
            },
            style: TextButton.styleFrom(foregroundColor: Colors.red),
            child: const Text('删除'),
          ),
        ],
      ),
    );
  }

  void _showPresetDialog() {
    showDialog(
      context: context,
      builder: (ctx) => _PresetPickerDialog(
        tool: widget.tool,
        onSave: () {
          p.Provider.of<AppState>(context, listen: false).loadProviders();
        },
      ),
    );
  }
}

/// Provider 行
class _ProviderRow extends StatelessWidget {
  final Provider provider;
  final VoidCallback onSwitch;
  final VoidCallback onEdit;
  final VoidCallback onDelete;

  const _ProviderRow({
    required this.provider,
    required this.onSwitch,
    required this.onEdit,
    required this.onDelete,
  });

  @override
  Widget build(BuildContext context) {
    final isDark = Theme.of(context).brightness == Brightness.dark;
    final cardColor = isDark ? const Color(0xFF2D2D2D) : const Color(0xFFF5F5F5);

    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 10),
      decoration: BoxDecoration(
        color: cardColor,
        borderRadius: BorderRadius.circular(8),
      ),
      child: Row(
        children: [
          // 激活状态指示
          InkWell(
            onTap: onSwitch,
            borderRadius: BorderRadius.circular(12),
            child: Icon(
              provider.isActive ? Icons.check_circle : Icons.circle_outlined,
              size: 20,
              color: provider.isActive ? Colors.green : Colors.grey[400],
            ),
          ),
          const SizedBox(width: 10),

          // 信息
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  children: [
                    Text(
                      provider.name,
                      style: TextStyle(
                        fontSize: 13,
                        fontWeight: provider.isActive ? FontWeight.w600 : FontWeight.normal,
                      ),
                    ),
                    if (provider.isActive) ...[
                      const SizedBox(width: 6),
                      Container(
                        padding: const EdgeInsets.symmetric(horizontal: 4, vertical: 1),
                        decoration: BoxDecoration(
                          color: Colors.green.withValues(alpha: 0.12),
                          borderRadius: BorderRadius.circular(3),
                        ),
                        child: const Text(
                          '使用中',
                          style: TextStyle(fontSize: 9, fontWeight: FontWeight.w500, color: Colors.green),
                        ),
                      ),
                    ],
                  ],
                ),
                const SizedBox(height: 2),
                Row(
                  children: [
                    if (provider.model.isNotEmpty) ...[
                      Text(
                        provider.model,
                        style: TextStyle(fontSize: 11, color: Colors.grey[500]),
                      ),
                      const SizedBox(width: 6),
                    ],
                    Text(
                      provider.maskedApiKey,
                      style: TextStyle(
                        fontSize: 10,
                        fontFamily: 'Consolas',
                        color: Colors.grey[400],
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),

          // 操作按钮
          OutlinedButton(
            onPressed: onEdit,
            style: OutlinedButton.styleFrom(
              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
              textStyle: const TextStyle(fontSize: 12),
              minimumSize: Size.zero,
            ),
            child: const Text('编辑'),
          ),
          const SizedBox(width: 8),
          InkWell(
            onTap: onDelete,
            borderRadius: BorderRadius.circular(4),
            child: Icon(Icons.delete_outline, size: 16, color: Colors.red[300]),
          ),
        ],
      ),
    );
  }
}

/// Provider 添加/编辑表单对话框
class _ProviderFormDialog extends StatefulWidget {
  final String tool;
  final String toolName;
  final Provider? provider;
  final VoidCallback onSave;

  const _ProviderFormDialog({
    required this.tool,
    required this.toolName,
    this.provider,
    required this.onSave,
  });

  @override
  State<_ProviderFormDialog> createState() => _ProviderFormDialogState();
}

class _ProviderFormDialogState extends State<_ProviderFormDialog> {
  late TextEditingController _nameCtrl;
  late TextEditingController _apiKeyCtrl;
  late TextEditingController _apiBaseCtrl;
  late TextEditingController _modelCtrl;
  late TextEditingController _extraEnvCtrl;
  late TextEditingController _notesCtrl;
  late TextEditingController _haikuCtrl;
  late TextEditingController _sonnetCtrl;
  late TextEditingController _opusCtrl;
  String? _errorMessage;

  bool get _isEditing => widget.provider != null;

  @override
  void initState() {
    super.initState();
    final pr = widget.provider;
    _nameCtrl = TextEditingController(text: pr?.name ?? '');
    _apiKeyCtrl = TextEditingController(text: pr?.apiKey ?? '');
    _apiBaseCtrl = TextEditingController(text: pr?.apiBase ?? '');
    _modelCtrl = TextEditingController(text: pr?.model ?? '');
    _extraEnvCtrl = TextEditingController(text: pr?.extraEnv ?? '{}');
    _notesCtrl = TextEditingController(text: pr?.notes ?? '');

    final extra = pr?.extraEnvDict;
    _haikuCtrl = TextEditingController(text: extra?['ANTHROPIC_DEFAULT_HAIKU_MODEL'] ?? '');
    _sonnetCtrl = TextEditingController(text: extra?['ANTHROPIC_DEFAULT_SONNET_MODEL'] ?? '');
    _opusCtrl = TextEditingController(text: extra?['ANTHROPIC_DEFAULT_OPUS_MODEL'] ?? '');
  }

  @override
  void dispose() {
    _nameCtrl.dispose();
    _apiKeyCtrl.dispose();
    _apiBaseCtrl.dispose();
    _modelCtrl.dispose();
    _extraEnvCtrl.dispose();
    _notesCtrl.dispose();
    _haikuCtrl.dispose();
    _sonnetCtrl.dispose();
    _opusCtrl.dispose();
    super.dispose();
  }

  Future<void> _save() async {
    setState(() => _errorMessage = null);

    final formData = ProviderFormData(
      name: _nameCtrl.text.trim(),
      tool: widget.tool,
      apiKey: _apiKeyCtrl.text.trim(),
      apiBase: _apiBaseCtrl.text.trim(),
      model: _modelCtrl.text.trim(),
      extraEnv: _extraEnvCtrl.text.trim(),
      icon: widget.provider?.icon,
      iconColor: widget.provider?.iconColor,
      notes: _notesCtrl.text.isEmpty ? null : _notesCtrl.text,
      category: widget.provider?.category,
      presetId: widget.provider?.presetId,
      haikuModel: _haikuCtrl.text.trim(),
      sonnetModel: _sonnetCtrl.text.trim(),
      opusModel: _opusCtrl.text.trim(),
    );

    try {
      final appState = p.Provider.of<AppState>(context, listen: false);
      if (_isEditing) {
        await appState.providerService.updateProvider(widget.provider!.id!, formData);
      } else {
        await appState.providerService.createProvider(formData);
      }
      widget.onSave();
      if (mounted) Navigator.pop(context);
    } catch (e) {
      setState(() => _errorMessage = e.toString());
    }
  }

  @override
  Widget build(BuildContext context) {
    return Dialog(
      child: SizedBox(
        width: 500,
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            // 标题
            Padding(
              padding: const EdgeInsets.all(16),
              child: Row(
                children: [
                  Text(
                    _isEditing ? '编辑 ${widget.toolName} 供应商' : '添加 ${widget.toolName} 供应商',
                    style: const TextStyle(fontSize: 16, fontWeight: FontWeight.w600),
                  ),
                  const Spacer(),
                  IconButton(
                    icon: const Icon(Icons.close, size: 18),
                    onPressed: () => Navigator.pop(context),
                  ),
                ],
              ),
            ),
            const Divider(height: 1),

            // 表单
            Flexible(
              child: SingleChildScrollView(
                padding: const EdgeInsets.all(16),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    _FormField(label: '名称 *', controller: _nameCtrl, hint: '如：官方 Claude'),
                    _FormField(label: 'API Key *', controller: _apiKeyCtrl, hint: 'sk-xxx', obscure: true),
                    _FormField(label: 'API 端点', controller: _apiBaseCtrl, hint: '空=使用官方默认端点'),
                    _FormField(label: '主模型', controller: _modelCtrl, hint: '空=使用默认模型'),
                    if (widget.tool == 'claude_code') ...[
                      _FormField(label: 'Haiku 模型', controller: _haikuCtrl, hint: '空=默认'),
                      _FormField(label: 'Sonnet 模型', controller: _sonnetCtrl, hint: '空=默认'),
                      _FormField(label: 'Opus 模型', controller: _opusCtrl, hint: '空=默认'),
                    ],
                    _FormField(label: '额外环境变量 (JSON)', controller: _extraEnvCtrl, hint: '{}', maxLines: 3),
                    _FormField(label: '备注', controller: _notesCtrl, hint: '可选'),
                    if (_errorMessage != null)
                      Padding(
                        padding: const EdgeInsets.only(top: 8),
                        child: Text(_errorMessage!, style: const TextStyle(color: Colors.red, fontSize: 12)),
                      ),
                  ],
                ),
              ),
            ),

            const Divider(height: 1),
            Padding(
              padding: const EdgeInsets.all(16),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.end,
                children: [
                  TextButton(
                    onPressed: () => Navigator.pop(context),
                    child: const Text('取消'),
                  ),
                  const SizedBox(width: 8),
                  ElevatedButton(
                    onPressed: _nameCtrl.text.trim().isEmpty || _apiKeyCtrl.text.trim().isEmpty
                        ? null
                        : _save,
                    child: Text(_isEditing ? '保存' : '添加'),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}

class _FormField extends StatelessWidget {
  final String label;
  final TextEditingController controller;
  final String hint;
  final bool obscure;
  final int maxLines;

  const _FormField({
    required this.label,
    required this.controller,
    this.hint = '',
    this.obscure = false,
    this.maxLines = 1,
  });

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.only(bottom: 12),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(label, style: const TextStyle(fontSize: 12, fontWeight: FontWeight.w500)),
          const SizedBox(height: 4),
          TextField(
            controller: controller,
            obscureText: obscure,
            maxLines: obscure ? 1 : maxLines,
            style: const TextStyle(fontSize: 13),
            decoration: InputDecoration(
              hintText: hint,
              hintStyle: TextStyle(fontSize: 13, color: Colors.grey[400]),
              isDense: true,
              contentPadding: const EdgeInsets.symmetric(horizontal: 10, vertical: 8),
              border: const OutlineInputBorder(),
            ),
          ),
        ],
      ),
    );
  }
}

/// 预设选择对话框
class _PresetPickerDialog extends StatefulWidget {
  final String tool;
  final VoidCallback onSave;

  const _PresetPickerDialog({required this.tool, required this.onSave});

  @override
  State<_PresetPickerDialog> createState() => _PresetPickerDialogState();
}

class _PresetPickerDialogState extends State<_PresetPickerDialog> {
  ProviderPreset? _selectedPreset;
  final _apiKeyCtrl = TextEditingController();
  String? _errorMessage;

  List<ProviderPreset> get _filteredPresets {
    return ProviderPreset.presets.where((p) => p.tool == widget.tool).toList();
  }

  @override
  void dispose() {
    _apiKeyCtrl.dispose();
    super.dispose();
  }

  Future<void> _addFromPreset() async {
    if (_selectedPreset == null) return;
    setState(() => _errorMessage = null);

    final formData = ProviderFormData(
      name: _selectedPreset!.name,
      tool: _selectedPreset!.tool,
      apiKey: _apiKeyCtrl.text.trim(),
      apiBase: _selectedPreset!.apiBase,
      model: _selectedPreset!.defaultModel,
      icon: _selectedPreset!.icon,
      iconColor: _selectedPreset!.iconColor,
      presetId: _selectedPreset!.id,
    );

    try {
      final appState = p.Provider.of<AppState>(context, listen: false);
      await appState.providerService.createProvider(formData);
      widget.onSave();
      if (mounted) Navigator.pop(context);
    } catch (e) {
      setState(() => _errorMessage = e.toString());
    }
  }

  @override
  Widget build(BuildContext context) {
    return Dialog(
      child: SizedBox(
        width: 450,
        height: 400,
        child: Column(
          children: [
            Padding(
              padding: const EdgeInsets.all(16),
              child: Row(
                children: [
                  const Text('从预设添加', style: TextStyle(fontSize: 16, fontWeight: FontWeight.w600)),
                  const Spacer(),
                  IconButton(icon: const Icon(Icons.close, size: 18), onPressed: () => Navigator.pop(context)),
                ],
              ),
            ),
            const Divider(height: 1),
            Expanded(
              child: ListView.builder(
                itemCount: _filteredPresets.length,
                itemBuilder: (context, i) {
                  final preset = _filteredPresets[i];
                  final isSelected = _selectedPreset?.id == preset.id;
                  return ListTile(
                    selected: isSelected,
                    title: Text(preset.name, style: const TextStyle(fontSize: 13, fontWeight: FontWeight.w500)),
                    subtitle: Text(
                      preset.apiBase.isEmpty ? '官方默认端点' : preset.apiBase,
                      style: TextStyle(fontSize: 11, color: Colors.grey[500]),
                    ),
                    trailing: Row(
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        Text(preset.defaultModel, style: TextStyle(fontSize: 11, color: Colors.grey[500])),
                        if (isSelected) ...[
                          const SizedBox(width: 8),
                          const Icon(Icons.check, color: Colors.green, size: 16),
                        ],
                      ],
                    ),
                    onTap: () => setState(() => _selectedPreset = preset),
                  );
                },
              ),
            ),
            if (_selectedPreset != null) ...[
              const Divider(height: 1),
              Padding(
                padding: const EdgeInsets.all(12),
                child: Row(
                  children: [
                    Expanded(
                      child: TextField(
                        controller: _apiKeyCtrl,
                        obscureText: true,
                        style: const TextStyle(fontSize: 13),
                        decoration: const InputDecoration(
                          hintText: '输入 API Key',
                          isDense: true,
                          contentPadding: EdgeInsets.symmetric(horizontal: 10, vertical: 8),
                          border: OutlineInputBorder(),
                        ),
                      ),
                    ),
                    const SizedBox(width: 8),
                    ElevatedButton(
                      onPressed: _apiKeyCtrl.text.trim().isEmpty ? null : _addFromPreset,
                      child: const Text('添加'),
                    ),
                  ],
                ),
              ),
            ],
            if (_errorMessage != null)
              Padding(
                padding: const EdgeInsets.symmetric(horizontal: 12),
                child: Text(_errorMessage!, style: const TextStyle(color: Colors.red, fontSize: 12)),
              ),
          ],
        ),
      ),
    );
  }
}
